Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -624,8 +664,15 @@ void NetworkStorageManager::persist(const WebCore::ClientOrigin& origin, Complet
     if (persistedFilePath(origin).isEmpty())
         return completionHandler(false);
 
-    // FIXME: add heuristics to decide if origin can be persisted and write persited file if it can.
-    return completionHandler(false);
+    bool shouldFetchRegistrableDomains = m_persistCompletionHandlers.isEmpty();
+    m_persistCompletionHandlers.append({ origin, WTFMove(completionHandler) });
+    if (!shouldFetchRegistrableDomains)
```

Comment:
We probably need to fetch these domains only once since they are static info.
This will make it clearer that these domains are exempt from ITP classification, not that they are classified as non trackers.